### ISSUE TITLE
Add functionality for passing previous-image flag to analyzer

### ISSUE THREAD

**Issue by importhuman** [buildpacks/pack#1275]
### Description

#1198 added the previous-image flag to creator. According to #897, it needs to be added to analyzer as well. 

### Additional context
- [x] This feature should be documented somewhere

**Comment by importhuman** [comment]
I'm a bit stuck on what exactly the parameter for analyzer should be. For instance, in creator in #1198, we appended `-previous-image <prev-image>` to the flags. As per #897, analyzer needs this in the form of `analyzer <prev-image>`. What is this parameter exactly that is being passed? LifecycleImage? 
cc: @jromero

**Comment by aemengo** [comment]
@importhuman I'm having trouble following the question as written. In https://github.com/buildpacks/pack/pull/1198, you implemented the functionality with the following line of code [here](https://github.com/buildpacks/pack/blob/main/internal/build/lifecycle_execution.go#L217):

```go
...
   flags = append(flags, "-previous-image", l.opts.PreviousImage)
...
```

The analyzer would thus require the same parameter: `l.opts.PreviousImage`. Does this help?

**Comment by importhuman** [comment]
@aemengo yeah it does, I was initially thinking that creator already takes image as argument, so if analyzer also takes an argument, do I modify that or can I pass an extra parameter. I'll try appending the previous-image to args and make an initial PR.

**Comment by aemengo** [comment]
@importhuman Oh I understand now. Thank you for your patience with me, It's a good question.

As @natalieparellano explained to us in this [slack.buildpacks.io](https://buildpacks.slack.com/archives/CD61YAG69/p1624904455090600?thread_ts=1624553333.080000&cid=CD61YAG69) thread: analyzer already takes an argument so:

```
analyzer <image>
```

should become

```
analyzer <prev-image>
```

And it makes sense because the analyzer phase is in fact looking for optimizations from the _previous_ build of your application image.

**Comment by importhuman** [comment]
@aemengo so is the parameter to be changed `l.opts.Image` or `l.opts.LifecycleImage`? 😅

**Comment by aemengo** [comment]
@importhuman Ah. Well, I'm not quite so sure myself. There's an opportunity to learn here that I don't want to steal from you. 🙂

**Comment by importhuman** [comment]
@aemengo that's fair I suppose 😂 looking at tests in `internal/build/lifecycle_execution_test.go`, analyzer runs the phase with lifecycle image whether publish is true or false, so based on that I reckon tinkering with it is not the correct approach, rather the image should be modified. I'll work on some tests in `lifecycle_execution_test.go` before the next commit.
